js/modules/k6/browser/common/network_manager.go

```diff
@@ -180,6 +180,7 @@ func (m *NetworkManager) emitRequestMetrics(req *Request) {
 	if state.Options.SystemTags.Has(k6metrics.TagURL) {
 		tags = handleURLTag(m.mi, req.URL(), req.method, tags)
 	}
+	tags = tags.With("resource_type", req.ResourceType())
```

Comment:
There are two things that I'm not sure of:

1. This isn't a system tag, but should it be?
2. This list can change at a moments notice by the chrome team, should we map them from the [chrome ResourceType](https://github.com/grafana/k6/blob/4847cef857bb4ac3a5f104a276496b18602f182e/vendor/github.com/chromedp/cdproto/network/types.go#L32-L49) to a k6 version of it. We would be able to detect if we stop matching on any of the types and log a warning, plus create a unit test to catch this as early as possible.

Reply:
> We would be able to detect if we stop matching on any of the types and log a warning, plus create a unit test to catch this as early as possible.

This sounds like the way to go! 👍🏻

Reply:
I've added a way to validate the ResourceTypes.